compiler-cli/src/dependencies.rs

```diff
@@ -678,6 +630,239 @@ fn same_requirements(
     Ok(left == right)
 }
 
+pub struct DependencyManagerConfig {
+    /// In `Prod` mode, dev dependencies are not considered during the executed operation.
+    /// Otherwise (`Dev` or `Lsp`), all dependencies are considered
+    pub mode: Mode,
```

Comment:
This has moved this internal mode value into the public interface now. What was your motivation for this? From what I can see it is never set to anything but `Dev`.

Reply:
Yeah you're right. I didn't intend to move it into the public interface, but to bundle configuration related data into a single struct, to reduce number of arguments passed around.

For now I'll remove `mode` from `DependencyManagerConfig`.

Reply:
I have also made `DependencyManager` private as it's not directly used outside of `dependencies`.